plugins/VstEffect/VstEffectControlDialog.h

```diff
@@ -67,7 +67,7 @@ class VstEffectControlDialog : public EffectControlDialog
 	PixmapButton * m_managePluginButton;
 	PixmapButton * m_savePresetButton;
 
-	QSharedPointer<VstPlugin> m_plugin;
+	QPointer<VstPlugin> m_plugin;
```

Comment:
Can we get rid of the Windows-specific behaviour instead? Maybe the bug this workaround was made for has been fixed.

Reply:
The workaround is very old(more than 5 years ago): https://github.com/LMMS/lmms/commit/1c9c76f3994a2054b5f4cb2f6b24c7f9b7fd1a10#diff-e3cda83bb57677b7ecd60d7e5e1fe92cR156
There have been so many changes after that(Qt version change, VST bug fixes, etc.) I think we should test whether the bug still exists.

Reply:
Removing the workaround, I can still reproduce the issue:
1. Enable "One instrument track window mode"
2. Create a project with at least two instruments
3. Add a VST effect to one of them
4. Open the other instrument's window
5. Open the first instrument's window again and show the VST effect dialog

The VST GUI will have disappeared.

I'm sure there's a better way to work around this, but the X11 embedding on Qt4 seems to be especially fragile and I'm not going to have access to a Linux machine again for a couple of months, so I didn't want to touch this any more. If you can produce something better, it would be very welcome.

Reply:
> Removing the workaround, I can still reproduce the issue:

Could you tell me your OS and Qt version you used? If I can reproduce that, I'll try to figure out what's going on.

Reply:
Windows 10 64-bit, Qt 5.4.2.

Reply:
I can reproduce the bug on GNU/Linux, too.
I found that the plugin widget is destroyed when `VstEffectControlDialog` is destroyed because of parenting. Simple unparenting seems to fix this.

Reply:
@DomClark Is it fine to remove eb560a780f8bb02cc4bbc02ca893eb1b850b9b47 from this PR and add new fix which uses unparenting?

Reply:
Sure, go ahead 😃

Reply:
Unparenting approach dosen't work well for XEmbed + Qt4. So I think I need to wrap that with QT version check macro if I do that.

Reply:
It seems like destroying parent window also destroys child windows on Windows. I guess it was the reason for the workaround.
https://msdn.microsoft.com/en-us/library/windows/desktop/ms632682(v=vs.85).aspx

Might it be better to use a separate helper class to handle `win32` embedding?